pandas/tseries/offsets.py

```diff
@@ -603,61 +641,76 @@ def next_bday(self):
         else:
             return BusinessDay(n=nb_offset)
 
-    @cache_readonly
-    def _get_daytime_flag(self):
-        if self.start == self.end:
-            raise ValueError('start and end must not be the same')
-        elif self.start < self.end:
-            return True
-        else:
-            return False
+    def _get_daytime_flag(self, start, end):
+        return start < end
 
-    def _next_opening_time(self, other):
+    def _next_opening_time(self, other, sign=1):
         """
```

Comment:
can you add a doc-string (e.g. the Parameters / Returns sections); ideally also type parameters as much as you can

Reply:
Updated